File: services-producer/src/todo/todo.model.ts
import { z } from 'zod'

export enum Status {
    ACTIVE = 'active',
    INACTIVE = 'inActive',
}

export interface Todo {
    id: string
    title: string
    description?: string
    done: boolean
    startDate: string
    endDate: string
    status: Status
}

export const paramsSchema = z.object({
    id: z.string(),
})

export const todoSchema = z.object({
    id: z.string(),
    title: z.string(),
    description: z.string().optional(),
    done: z.boolean(),
    startDate: z.string(),
    endDate: z.string(),
    status: z.nativeEnum(Status),
})

File: services-http/todo/todo.model.go
package todo

type ResponseData[T any] struct {
	Items    T     `json:"items"`
	Total    int64 `json:"total"`
	Page     int   `json:"page"`
	PageSize int   `json:"pageSize"`
}

type TaskQuery struct {
	Status   string `json:"status"`
	Page     int    `json:"page"`
	PageSize int    `json:"pageSize"`
	Sort     string `json:"sort"`
	Order    int    `json:"order"`
}

type Task struct {
	ID          string `json:"id" bson:"id"`
	Href        string `json:"href"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Done        bool   `json:"done"`
	StartDate   string `json:"startDate"`
	EndDate     string `json:"endDate"`
	Status      string `json:"status"`
}
